botnim/document_parser/dynamic_extractions/pdf_extraction/google_sheets_sync.py

```diff
@@ -0,0 +1,311 @@
+import logging
```

Comment:
Putting this here but it's a general design issue:

Coupling the PDF extraction code with the google spreadsheets code so tightly is not a good design. As we discussed, the input for the PDF extraction pipeline should be a CSV file & a directory of PDF files (+metadata). The output would be an updated CSV file.
We can even think of a simple contract where it receives a single path, containing:
- `input.csv`
- `<XXX>.pdf`
- `<XXX>.pdf.metadata.json`
And it outputs, in the same path `output.csv`

If there's code to pull/push the CSV file to anywhere in the cloud (be it google spreadsheets or a different method), it needs to be invisible to the PDF pipeline - and not embedded in it.